exercise/exercise_summary.py

```diff
@@ -126,10 +126,10 @@ def is_graded(self):
     def is_unofficial(self):
         return self.unofficial
 
-    def get_group(self):
+    def get_group(self) -> Optional[StudentGroup]:
         if self.submission_count > 0:
             s = self.submissions[0]
-            if s.submitters.count() > 0:
+            if s.submitters.exists():
                 return StudentGroup.get_exact(
                     self.exercise.course_instance,
                     s.submitters.all()
```

Comment:
This could be improved based on this:
https://docs.djangoproject.com/en/3.2/topics/db/optimization/#don-t-overuse-count-and-exists

If there are submitters, we could run this with only one database query like this:
```suggestion
            submitters = s.submitters.all()
            if submitters:
                return StudentGroup.get_exact(
                    self.exercise.course_instance,
                    submitters,
```
If there are no submitters, I guess using `exists()` to speed up that case is not really worthwhile.

Reply:
That wouldn't do anything (I tried). The submitters of the submission are already prefetched by SubmissionManager. `exists` checks the result cache before executing a new query.